### PR TITLE
Stabilise email-result handling

### PR DESCRIPTION
**Pull request by myieye**
This load function is fragile. It was tricky setting it up and when I changed to CSR (for experimentation) it stopped working.

I think we're generally not supposed to use stores in load functions, but I also think it's fine if it's only client-side. So, I think this is a good change, but I'm putting it in a PR so it's more visible.

### CODE REVIEW

File: frontend/src/routes/(authenticated)/user/+page.ts [hunk 1]
```diff
@@ -16,9 +16,13 @@ export const load = (({ url }) => {
   const emailResult = url.searchParams.get('emailResult') as EmailResult | null;
   if (emailResult) {
     if (!EMAIL_RESULTS.includes(emailResult)) throw new Error(`Invalid emailResult: ${emailResult}.`);
-    if (browser) void goto(`${url.pathname}`, { replaceState: true });
+    if (browser) {
+      void goto(`${url.pathname}`, { replaceState: true });
```
Comment (hahn-kev) [inline frontend/src/routes/(authenticated)/user/+page.ts:20]: are we sure this isn't what's causing issues? this is just to remove the search param from the url to keep it pretty right? If that's the case maybe we should just remove it.

Comment (myieye) [reply frontend/src/routes/(authenticated)/user/+page.ts]: Very good idea 😆.
I just removed the troublesome, unnecessarily complicated, minor UX feature.

**Comment by myieye** [comment]
I also greatly simplified closing the banner/notification.

**Comment by myieye** [comment]
@hahn-kev feel free to review again, but it seemed pretty safe to merge and there are a lot of PRs open, so I just went for it.

**Pull request merged**